### PR TITLE
♻️ Rx Refactor NoWorkspaceViewModel

### PR DESCRIPTION
**Pull request by khoi**
## What's this?
Rx Refactor NoWorkspaceViewModel

### Relationships
Part of #3352

## Why do we want this?
To make our lives easier.

## How is it done?
By rx-ifying stuff

### Why not another way?
Could be simplified when #3490 is merged.

### Side effects
None

## Review hints
None

## :squid: Permissions
🐙

### CODE REVIEW

**Comment by khoi** [comment]
WIP until `RxAction`'s `.Execute()` works properly

File: Toggl.Foundation.MvvmCross/ViewModels/NoWorkspaceViewModel.cs [hunk 1]
```diff
@@ -20,28 +24,40 @@ public sealed class NoWorkspaceViewModel : MvxViewModel
...
+            Ensure.Argument.IsNotNull(schedulerProvider, nameof(schedulerProvider));
 
             this.dataSource = dataSource;
             this.accessRestrictionStorage = accessRestrictionStorage;
             this.navigationService = navigationService;
             this.interactorFactory = interactorFactory;
+
+            CreateWorkspaceWithDefaultName = UIAction.FromAsync(createWorkspaceWithDefaultName);
+            TryAgain = UIAction.FromAsync(tryAgain);
+            IsLoading = isLoading.AsDriver(false, schedulerProvider);
```
Comment (simonrozsival) [inline Toggl.Foundation.MvvmCross/ViewModels/NoWorkspaceViewModel.cs]: Add an explicit name to the `true` literal please!
```suggestion
            IsLoading = isLoading.AsDriver(onErrorJustReturn: false, schedulerProvider: schedulerProvider);
```

File: Toggl.Daneel/ViewControllers/NoWorkspaceViewController.cs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 ﻿using System.Reactive.Linq;
 using CoreGraphics;
+using MvvmCross.Binding.BindingContext;
```
Comment (simonrozsival) [inline Toggl.Daneel/ViewControllers/NoWorkspaceViewController.cs]: Is this needed?

File: Toggl.Foundation.MvvmCross/ViewModels/NoWorkspaceViewModel.cs [hunk 3]
```diff
@@ -20,28 +21,40 @@ public sealed class NoWorkspaceViewModel : MvxViewModel
...
+            Ensure.Argument.IsNotNull(schedulerProvider, nameof(schedulerProvider));
 
             this.dataSource = dataSource;
             this.accessRestrictionStorage = accessRestrictionStorage;
             this.navigationService = navigationService;
             this.interactorFactory = interactorFactory;
+
+            CreateWorkspaceWithDefaultName = UIAction.FromAsync(createWorkspaceWithDefaultName);
+            TryAgain = UIAction.FromAsync(tryAgain);
+            IsLoading = isLoading.AsDriver(onErrorJustReturn: false, schedulerProvider: schedulerProvider);
```
Comment (kornelijepetak) [inline Toggl.Foundation.MvvmCross/ViewModels/NoWorkspaceViewModel.cs]: No need for repeated name for `schedulerProvider`.

File: Toggl.Foundation.Tests/MvvmCross/ViewModels/NoWorkspaceViewModelTests.cs [hunk 4]
```diff
@@ -142,7 +146,8 @@ public async Task StartsAndStopsLoading()
                 ViewModel.IsLoading.Subscribe(observer);
                 InteractorFactory.CreateDefaultWorkspace().Execute().Returns(Observable.Return(Unit.Default));
 
-                await ViewModel.CreateWorkspaceWithDefaultName();
+                await ViewModel.CreateWorkspaceWithDefaultName.Execute();
+                TestScheduler.Start();
```
Comment (kornelijepetak) [inline Toggl.Foundation.Tests/MvvmCross/ViewModels/NoWorkspaceViewModelTests.cs]: Across our tests I've mostly seen `TestScheduler.Start()` grouped with 3rd A in (AAA).

File: Toggl.Foundation.MvvmCross/ViewModels/NoWorkspaceViewModel.cs (hunk 3 above)
Comment (khoi) [reply Toggl.Foundation.MvvmCross/ViewModels/NoWorkspaceViewModel.cs]: we agreed to leave it as is since having only one function label requires C# > 7.2

**Pull request merged**